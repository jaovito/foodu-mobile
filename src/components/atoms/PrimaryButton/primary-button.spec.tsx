import {render, fireEvent} from '@test-utils';
import {PrimaryButton} from '.';

describe('PrimaryButton Tests', () => {
  it('should render correctly', () => {
    const button = render(<PrimaryButton iconName="minus" />);

    expect(button).toBeDefined();
  });

  it('should be able to called on press', () => {
    const onPress = jest.fn();
    const button = render(<PrimaryButton iconName="minus" onPress={onPress} />);

    fireEvent.press(button.getByTestId('PrimaryButtonID'));

    expect(onPress).toBeCalled();
  });

  it('should be able to match with snapshot without onPress', () => {
    const button = render(<PrimaryButton iconName="minus" />).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should be able to match with snapshot with onPress', () => {
    const onPress = jest.fn();

    const button = render(
      <PrimaryButton iconName="minus" onPress={onPress} />,
    ).toJSON();

    expect(button).toMatchSnapshot();
  });
});
