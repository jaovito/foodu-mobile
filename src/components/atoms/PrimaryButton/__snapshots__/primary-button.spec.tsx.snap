// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton Tests should be able to match with snapshot with onPress 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FE8C00",
      "borderRadius": 31,
      "height": 63,
      "justifyContent": "center",
      "opacity": 1,
      "width": 63,
    }
  }
  testID="PrimaryButtonID"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 12,
        },
        [
          {
            "color": "#FFFFFF",
            "fontSize": 31,
          },
        ],
        {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`PrimaryButton Tests should be able to match with snapshot without onPress 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FE8C00",
      "borderRadius": 31,
      "height": 63,
      "justifyContent": "center",
      "opacity": 1,
      "width": 63,
    }
  }
  testID="PrimaryButtonID"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 12,
        },
        [
          {
            "color": "#FFFFFF",
            "fontSize": 31,
          },
        ],
        {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</View>
`;
